lib/lib.misc.php

```diff
@@ -1319,26 +1344,41 @@ function eventlog($type, $dataid, $action, $cid = null, $json = null, $id = null
 	}
 
 	// Look up external/API ID from various sources.
-	if ( $id===null ) $id = rest_extid($type, $dataid);
+	if ( $ids===null ) {
+		$ids = array_map(function($dataid) use ($type) {
+			return rest_extid($type, $dataid);
+		}, $dataids);
+	}
 
-	if ( $id===null && $json!==null ) {
+	if ( $ids===null && $json!==null ) {
```

Comment:
can $ids be null at this point? it's rather that single entries in the array are null, right? Do we have to do the check here for every id instead?

Reply:
When `$json` is not null, it can only be `[null]`. So I have changed this.